app/docs/enterprise/0.31-x/oidc-auth0.md

```diff
@@ -0,0 +1,41 @@
+---
+title: OpenID Connect with Auth0
+class: page-install-method
+---
+
+This guide covers an example OpenID Connect plugin configuration to authenticate headless services using Auth0's identity provider.
+
+# Auth0 IDP configuration
+
+This configuration will use a [client credentials grant](https://auth0.com/docs/api-auth/tutorials/client-credentials) as it is non-interactive, and because we expect clients to authenticate on behalf of themselves, not an end-user. To do so, you will need to [create an Auth0 API](https://auth0.com/docs/apis#how-to-configure-an-api-in-auth0) and a [non-interactive client](https://auth0.com/docs/clients).
+
+## API configuration
+
+When creating your API, you will need to specify an Identifier. Using the URL that your service makes requests to is generally appropriate, so this will typically be the hostname/path combination configured as a service or API in Kong.
+
+After creating your API, you will also need to add the `openid` scope at `https://manage.auth0.com/#/apis/<API ID>/scopes`.
+
+## Client configuration
+
+You will need to authorize your client to access your API. Auth0 will prompt you to do so after client creation if you select the API you created previously from the client's Quick Start menu. After toggling the client to Authorized, expand its authorization settings and enable the `openid` scope.
+
+# Kong configuration
+
+If you have not done so already, [create a service](https://getkong.org/docs/0.13.x/admin-api/#add-service) to protect. The `url` configuration should match the Identifier you used when configuring Auth0.
+
+Add an OpenID plugin configuration following the example below. Auth0's token endpoint [requires passing the API identifier in the `audience` parameter](https://auth0.com/docs/api/authentication#client-credentials), which must be added as a custom argument:
```

Comment:
Ideally we'd show examples using the Admin GUI - perhaps you can at least note that this configuration can be done via Admin GUI too?

Reply:
Added mentions. Examples of the GUI are hard at present because it displays almost all of the (many) options for OIDC, and most aren't used, so any screenshots would be very large and consist of mostly whitespace.